### PR TITLE
Switch Download tables to Markdown from HTML

### PR DESCRIPTION
**Pull request by blaisb**
As discussed with @tjhei  a few weeks ago, it is preferable to make the tables in Markdown instead of keeping the raw HTML ones. I have taken the liberty of fixing all tables in the download page. I have furthermore updated the MAC OS version. However, we currently do not have the .dmg file of the latest release 9.6 on our mirror. @luca-heltai could you take care of that and after that i can make a PR to add the new link? :) 

The new rendering is much cleaner (IMO). It looks like this:
image

### CODE REVIEW

File: docs/current_release/download.md [hunk 1]
```diff
@@ -54,187 +54,20 @@ Development sources
...
-                      deal.II is also distributed in <b><a href="https://spack.io/">Spack</a></b>,
-                      a package manager for supercomputers, Linux and Mac OS.
-                      With Spack you can build packages with multiple versions, configurations, platforms, and compilers.
-                      See the
-                      <a href="https://github.com/dealii/dealii/wiki/deal.II-in-Spack">deal.II Spack Wiki page</a> for details.
-                    </p>
-                  </td>
-                </tr>
-              </table>
+| **Mac OS packages**        |  dealii-9.6.0-sonoma-arm64.dmg, see [github](https://github.com/dealii/dealii/releases/download/v9.6.0/dealii-9.6.0-sonoma-arm64-clang15.dmg)  <br>  (PGP signature file: [github](https://github.com/dealii/dealii/releases/download/v9.6.0/dealii-9.6.0-sonoma-arm64-clang15.dmg.asc)  <br> August 31, 2024, 1.71GB.  <br>  [Mac OSX Instructions](https://github.com/dealii/dealii/wiki/MacOSX) |
```
Comment (tjhei) [inline docs/current_release/download.md:57]:
```suggestion
| **Mac OS packages**        |  dealii-9.6.0-sonoma-arm64.dmg, see [github](https://github.com/dealii/dealii/releases/download/v9.6.0/dealii-9.6.0-sonoma-arm64-clang15.dmg)  <br>  (PGP signature file: [github](https://github.com/dealii/dealii/releases/download/v9.6.0/dealii-9.6.0-sonoma-arm64-clang15.dmg.asc)) <br> August 31, 2024, 1.71GB.  <br>  [Mac OSX Instructions](https://github.com/dealii/dealii/wiki/MacOSX) |
```

**Review by tjhei — APPROVED** [review]
awesome, thank you!

**Review by masterleinad — APPROVED** [review]

**Comment by luca-heltai** [comment]
Actually, I think only @tamiko can upload to the mirror. Or can I?

**Comment by tamiko** [comment]
@luca-heltai The mirror is now synonymous with our webserver - but yes, currently I need to upload manually.

**Pull request merged**